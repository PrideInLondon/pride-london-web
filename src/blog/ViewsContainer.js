import React from 'react'
import PropTypes from 'prop-types'
import Slider from 'react-slick'
import { Container, Row, Column } from '../components/grid'
import ViewsCard from './ViewsCard'
import { CardContainer, ViewsTitle } from './ViewsContainer.styles'
import { settings } from './slickSettings'

class ViewsContainer extends React.Component {
  // Methods here to prevent horizontal scrolling
  // while swiping with react-slick, see
  // https://github.com/akiran/react-slick/issues/1240
  componentDidMount() {
    window.addEventListener('touchstart', this.touchStart)
    window.addEventListener('touchmove', this.preventTouch, { passive: false })
  }

  componentWillUnmount() {
    window.removeEventListener('touchstart', this.touchStart)
    window.removeEventListener('touchmove', this.preventTouch, {
      passive: false,
    })
  }

  touchStart(e) {
    this.firstClientX = e.touches[0].clientX
    this.firstClientY = e.touches[0].clientY
  }

  preventTouch(e) {
    const minValue = 5 // threshold

    this.clientX = e.touches[0].clientX - this.firstClientX
    this.clientY = e.touches[0].clientY - this.firstClientY

    // Vertical scrolling does not work when you start swiping horizontally.
    if (Math.abs(this.clientX) > minValue) {
      e.preventDefault()
      e.returnValue = false
      return false
    }
  }

  render() {
    const { views } = this.props
    return (
      <section>
        <Container>
          <Row>
            <Column width={1}>
              <CardContainer>
                <ViewsTitle isLight isCentered>
                  Views
                </ViewsTitle>
                <Slider {...settings}>
                  {views.map(view => (
                    <ViewsCard {...view} key={view.id} />
                  ))}
                </Slider>
              </CardContainer>
            </Column>
          </Row>
        </Container>
      </section>
    )
  }
}

ViewsContainer.propTypes = {
  views: PropTypes.arrayOf(PropTypes.shape({})),
}

ViewsContainer.defaultProps = {
  views: [],
}

export default ViewsContainer
