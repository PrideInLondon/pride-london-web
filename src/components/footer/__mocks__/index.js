export const mockSponsors = {
  allContentfulSponsor: {
    edges: [
      {
        node: {
          id: 'a1bc90c0-23c7-51e1-8ea2-4bce5224df33',
          sponsorName: 'Blue Sun',
          sponsorUrl: 'https://media.giphy.com/media/UvwI1X7XkbXq0/giphy.gif',
          sponsorLogo: {
            sizes: {
              src:
                '//images.ctfassets.net/n2o4hgsv6wcx/12nAORbw80eC666kck4QgU/168f3efbd9c99f54c96cab367ac73943/4822573655_b22cb18c05.jpg?w=800&h=168&q=90',
            },
          },
          sponsorLevel: 'Headline',
        },
      },
      {
        node: {
          id: '1e2d4e9d-1684-5657-a7b6-3c6a5e1e96c1',
          sponsorName: 'Virtucon',
          sponsorUrl: 'https://m.popkey.co/666ded/1VRvJ.gif',
          sponsorLogo: {
            sizes: {
              src:
                '//images.ctfassets.net/n2o4hgsv6wcx/2AfZNrit52m4WUm4IKUOOE/21323f2171228d475743a62938778c53/ed2cf58359281c503c8fd2d56992b0a9.png?w=800&h=168&q=90',
            },
          },
          sponsorLevel: 'Headline',
        },
      },
      {
        node: {
          id: '6bace2aa-7fa9-5969-8e0f-e9170d7f015d',
          sponsorName: 'Weyland-Yutani',
          sponsorUrl: 'http://alienanthology.wikia.com/wiki/Weyland-Yutani',
          sponsorLogo: {
            sizes: {
              src:
                '//images.ctfassets.net/n2o4hgsv6wcx/3DzjJqij9m4Ka0Q8IS8sMQ/66d8f571a8ac73b550ff9ae4460834cf/e44d54b01d679a151f570ce9c7d85cb3.png?w=800&h=168&q=90',
            },
          },
          sponsorLevel: 'Gold',
        },
      },
      {
        node: {
          id: '73cee0dd-94a9-5426-b52b-35ce3214ad4d',
          sponsorName: 'Abstergo',
          sponsorUrl: 'http://www.abstergo.com/',
          sponsorLogo: {
            sizes: {
              src:
                '//images.ctfassets.net/n2o4hgsv6wcx/1hr2YDhECsMUeu0KMqsge2/0b6ca7913c2ce37bf5d077e716540e8a/Abstergo_industries_by_vesferatu-d39vask.gif?w=800&h=168&q=90',
            },
          },
          sponsorLevel: 'Headline',
        },
      },
      {
        node: {
          id: '9f0f29b2-b53a-5b13-b0dd-d048041f8da5',
          sponsorName: 'Omni Consumer Products',
          sponsorUrl: 'http://omniconsumerproductscorporation.com/',
          sponsorLogo: {
            sizes: {
              src:
                '//images.ctfassets.net/n2o4hgsv6wcx/5yhojSLzLqgaSgAYMaAokW/63acd88bb36bcf035f823952eae9e1f6/kisspng-robocop-omni-consumer-products-ed-209-logo-cyborg-robocop-5acd74ab7e47c9.5852170415234141875173.jpg?w=800&h=168&q=90',
            },
          },
          sponsorLevel: 'Gold',
        },
      },
      {
        node: {
          id: '760bcc60-2704-54b5-8555-da4adab9f394',
          sponsorName: 'Testy Badger',
          sponsorUrl: 'https://red-badger.com/',
          sponsorLogo: {
            sizes: {
              src:
                '//images.ctfassets.net/n2o4hgsv6wcx/13iikTHlzcyA4oquoaWW4A/8e9e96aee3d35b5367b964e265894943/RB_Screen_Logos_For_Guardian.jpg?w=800&h=168&q=90',
            },
          },
          sponsorLevel: 'Headline',
        },
      },
      {
        node: {
          id: '5f4a1bb1-6c2d-502a-b47f-ea357d8103e8',
          sponsorName: 'Barclays',
          sponsorUrl: 'https://www.barclays.co.uk/',
          sponsorLogo: {
            sizes: {
              src:
                '//images.ctfassets.net/n2o4hgsv6wcx/5R7cc2vejSyq6qquSqGiyk/7536e8a3e4e5663fa1dbec357603728a/Barclays_Logo.png?w=800&h=168&q=90',
            },
          },
          sponsorLevel: 'Headline',
        },
      },
      {
        node: {
          id: 'd5fde3eb-032f-513e-8852-26e6b02b5d2d',
          sponsorName: 'Mayor of London',
          sponsorUrl: 'https://www.london.gov.uk/about-us/mayor-london',
          sponsorLogo: {
            sizes: {
              src:
                '//images.ctfassets.net/n2o4hgsv6wcx/4rIPcvrYGk8uE24oWI08Yi/e70dc626716b5fb59efa22a4b7da263f/MayorOfLondon_Logo.png?w=800&h=168&q=90',
            },
          },
          sponsorLevel: 'Headline',
        },
      },
      {
        node: {
          id: '6f20fdff-3cc4-54dc-9fdd-850de5a7f471',
          sponsorName: 'Tesco',
          sponsorUrl: 'https://www.tesco.com/',
          sponsorLogo: {
            sizes: {
              src:
                '//images.ctfassets.net/n2o4hgsv6wcx/E59pqZoWxamyWW8caocm2/975a6e54d6aeac70d452e303e8422b8c/Tesco_Logo.png?w=800&h=168&q=90',
            },
          },
          sponsorLevel: 'Gold',
        },
      },
      {
        node: {
          id: 'e7f18314-f613-552f-9f07-031f14840914',
          sponsorName: 'CMS',
          sponsorUrl: 'https://cms.law/en/GBR/',
          sponsorLogo: {
            sizes: {
              src:
                '//images.ctfassets.net/n2o4hgsv6wcx/1vKEetL34E4ekk0EikWYm0/bad1e9109ae4fcbe4d138a650ece43cd/CMS_Logo.png?w=800&h=168&q=90',
            },
          },
          sponsorLevel: 'Gold',
        },
      },
    ],
  },
}
