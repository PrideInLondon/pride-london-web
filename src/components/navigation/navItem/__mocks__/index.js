export const mockData = {
  id: 'nav-learn',
  title: 'Learn',
  desc:
    'Cupcake ipsum dolor sit. Amet dragée chocolate donut. Jelly-o chupa chups liquorice chocolate.',
  url: '/',
  submenu: [
    {
      heading: 'Jelly-o chupa chups',
      links: [
        { title: 'Cupcake ipsum dolor sit', url: '/' },
        { title: 'Cupcake ipsum sit', url: '/' },
        { title: 'Cupcake dolor sit', url: '/' },
        { title: 'Cupcake ipsum dolor', url: '/' },
      ],
    },
    {
      heading: 'Icing gummi bears powder',
      links: [
        { title: 'Amet dragée donut', url: '/' },
        { title: 'Amet dragée chocolate donut', url: '/' },
      ],
    },
    {
      heading: 'Bear claw pastry bonbon',
      links: [
        { title: 'Carrot cake fruitcake', url: '/' },
        { title: 'Carrot cake muffin fruitcake', url: '/' },
      ],
    },
  ],
}
