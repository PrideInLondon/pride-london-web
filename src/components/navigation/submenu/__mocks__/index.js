export const mockData = {
  submenu: [
    {
      heading: 'Jelly-o chupa chups',
      links: [
        { title: 'Cupcake ipsum dolor sit', url: '/' },
        { title: 'Cupcake ipsum sit', url: '/' },
        { title: 'Cupcake dolor sit', url: '/' },
        { title: 'Cupcake ipsum dolor', url: '/' },
      ],
    },
  ],
  title: 'Lorem',
  url: '/ipsum',
  desc: 'lorem ipsum',
  id: 'lorem',
}
