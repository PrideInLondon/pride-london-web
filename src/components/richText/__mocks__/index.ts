import {
  Document,
  NodeData,
  Mark,
  TopLevelBlock,
  MARKS,
} from '@contentful/rich-text-types'
import logo from '../../../assets/logo.png'

// nodeType properties below ignored due to issue with Contentful types
// see https://github.com/contentful/rich-text/issues/123

const generateNode = ({
  data = {},
  marks = [],
  value,
  nodeType = 'paragraph',
}: {
  data?: NodeData
  marks?: Mark[]
  value: string
  nodeType?: string
}): TopLevelBlock => ({
  data,
  content: [
    {
      data: {},
      marks,
      value,
      nodeType: 'text',
    },
  ],
  // @ts-ignore
  nodeType,
})

export const paragraph: Document = {
  content: [
    generateNode({
      value:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
    }),
    generateNode({
      value:
        'Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem. Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur?',
    }),
  ],
  // @ts-ignore
  nodeType: 'document',
}

export const formatted: Document = {
  content: [
    generateNode({ value: 'This is some regular text.' }),
    generateNode({
      value: 'This is some bold text.',
      marks: [{ type: MARKS.BOLD }],
    }),
    generateNode({
      value: 'This is some italic text.',
      marks: [{ type: MARKS.ITALIC }],
    }),
    generateNode({
      value: 'This is some underlined text.',
      marks: [{ type: MARKS.UNDERLINE }],
    }),
  ],
  // @ts-ignore
  nodeType: 'document',
}

export const links: Document = {
  content: [
    {
      content: [
        generateNode({
          data: { uri: 'https://prideinlondon.org' },
          value: 'This is an external link',
          nodeType: 'hyperlink',
        }),
      ],
      // @ts-ignore
      nodeType: 'paragraph',
    },
    {
      content: [
        generateNode({
          data: { uri: '/about-us' },
          value: 'This is an internal link',
          nodeType: 'hyperlink',
        }),
      ],
      // @ts-ignore
      nodeType: 'paragraph',
    },
  ],
  // @ts-ignore
  nodeType: 'document',
}

const generateList = (nodeType: 'unordered' | 'ordered'): Document => ({
  content: [
    {
      content: [
        {
          // @ts-ignore
          content: [generateNode({ value: 'First item' })],
          // @ts-ignore
          nodeType: 'list-item',
        },
        {
          // @ts-ignore
          content: [generateNode({ value: 'Second item' })],
          // @ts-ignore
          nodeType: 'list-item',
        },
        {
          // @ts-ignore
          content: [generateNode({ value: 'Third item' })],
          // @ts-ignore
          nodeType: 'list-item',
        },
      ],
      // @ts-ignore
      nodeType: `${nodeType}-list`,
    },
  ],
  // @ts-ignore
  nodeType: 'document',
})

export const unorderedList = generateList('unordered')

export const orderedList = generateList('ordered')

export const quote: Document = {
  content: [
    {
      content: [
        generateNode({
          value:
            'At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis praesentium voluptatum deleniti atque corrupti quos dolores et quas molestias excepturi sint occaecati cupiditate non provident, similique sunt in culpa qui officia deserunt mollitia animi, id est laborum et dolorum fuga.',
        }),
      ],
      // @ts-ignore
      nodeType: 'blockquote',
    },
  ],
  // @ts-ignore
  nodeType: 'document',
}

const generateAnyLocale = (val: any) => ({ 'en-GB': val })

const contentfulImage = {
  image: generateAnyLocale({
    fields: { file: generateAnyLocale({ url: logo }) },
  }),
  altText: generateAnyLocale('consectetur adipiscing elit'),
}

export const video: Document = {
  content: [
    {
      data: {
        target: {
          sys: {
            type: 'Link',
            contentType: {
              sys: {
                id: 'video',
              },
            },
          },
          fields: {
            host: generateAnyLocale('youtube'),
            videoId: generateAnyLocale('9v34wjMlT2A'),
            caption: generateAnyLocale('Lorem ipsum dolor sit amet'),
            image: generateAnyLocale({
              fields: contentfulImage,
            }),
          },
        },
      },
      content: [],
      // @ts-ignore
      nodeType: 'embedded-entry-block',
    },
  ],
  // @ts-ignore
  nodeType: 'document',
}

export const image: Document = {
  content: [
    {
      data: {
        target: {
          sys: {
            type: 'Link',
            contentType: {
              sys: {
                id: 'image',
              },
            },
          },
          fields: contentfulImage,
        },
      },
      content: [],
      // @ts-ignore
      nodeType: 'embedded-entry-block',
    },
  ],
  // @ts-ignore
  nodeType: 'document',
}

export const multiImage: Document = {
  content: [
    {
      data: {
        target: {
          sys: {
            type: 'Link',
            contentType: {
              sys: {
                id: 'multiImage',
              },
            },
          },
          fields: {
            images: generateAnyLocale([
              { fields: contentfulImage },
              { fields: contentfulImage },
              { fields: contentfulImage },
            ]),
          },
        },
      },
      content: [],
      // @ts-ignore
      nodeType: 'embedded-entry-block',
    },
  ],
  // @ts-ignore
  nodeType: 'document',
}
