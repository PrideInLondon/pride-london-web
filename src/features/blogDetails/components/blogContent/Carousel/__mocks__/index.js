export default {
  data: {
    target: {
      fields: {
        carouselItems: {
          'en-GB': [
            {
              sys: {
                space: {
                  sys: {
                    type: 'Link',
                    linkType: 'Space',
                    id: 'n2o4hgsv6wcx',
                  },
                },
                id: 'c4qNhcpAIfewoQYwE8kcwoa',
                type: 'Asset',
                createdAt: '2018-05-16T15:57:12.149Z',
                updatedAt: '2018-05-16T15:57:12.149Z',
                environment: {
                  sys: {
                    id: 'master',
                    type: 'Link',
                    linkType: 'Environment',
                  },
                },
                revision: 1,
              },
              fields: {
                title: {
                  'en-GB': 'e2b675c430b3abb69b074c418afa6b4c',
                },
                file: {
                  'en-GB': {
                    url:
                      '//images.ctfassets.net/n2o4hgsv6wcx/4qNhcpAIfewoQYwE8kcwoa/534c8fd313fb8aa1673b580354ecf13c/e2b675c430b3abb69b074c418afa6b4c.jpg',
                    details: {
                      size: 68811,
                      image: {
                        width: 500,
                        height: 729,
                      },
                    },
                    fileName: 'e2b675c430b3abb69b074c418afa6b4c.jpg',
                    contentType: 'image/jpeg',
                  },
                },
              },
            },
          ],
        },
        title: {
          'en-GB': 'carousel with ponies',
        },
      },
    },
  },
  content: [],
  nodeType: 'embedded-entry-block',
}
