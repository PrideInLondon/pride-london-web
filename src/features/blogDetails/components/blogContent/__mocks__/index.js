export const notRecognizedComponent = {
  data: {
    target: {
      sys: {
        contentType: {
          sys: {
            id: 'not-recognized-component',
          },
        },
      },
    },
  },
}

export const assetHyperlinkNode = {
  data: {
    target: {
      fields: {
        file: {
          'en-GB': {
            url: 'example.com',
          },
        },
      },
    },
  },
}

export const urlHyperlinkNode = {
  data: {
    uri: 'example.com',
  },
}

export const imageNode = {
  data: {
    target: {
      fields: {
        file: {
          'en-GB': {
            url: 'example.com',
          },
        },
      },
    },
  },
}
