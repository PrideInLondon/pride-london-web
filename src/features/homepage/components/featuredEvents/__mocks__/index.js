export const mockEvents = [
  {
    node: {
      id: '123abc',
      name: 'Pride Quiz',
      startTime: '2019-06-05T20:00+01:00',
      endTime: '2019-06-05T23:00+01:00',
      eventPriceLow: 3,
      eventsListPicture: {
        title: 'img alt',
        fixed: {
          base64: 'abc',
          height: 235,
          width: 400,
          src: 'src',
          srcSet: 'srcset',
        },
      },
    },
  },
  {
    node: {
      id: '123abcd',
      name: 'Pride Quiz 2',
      startTime: '2019-06-05T20:00+01:00',
      endTime: '2019-06-05T23:00+01:00',
      eventPriceLow: 3,
      eventsListPicture: {
        title: 'img alt',
        fixed: {
          base64: 'abc',
          height: 235,
          width: 400,
          src: 'src',
          srcSet: 'srcset',
        },
      },
    },
  },
]
