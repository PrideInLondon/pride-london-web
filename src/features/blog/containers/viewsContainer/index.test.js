import React from 'react'
import toJSON from 'enzyme-to-json'
import { shallow, mount } from 'enzyme'
import { views } from './__mocks__'
import ViewsContainer from '.'

describe('ViewsContainer', () => {
  it('should render with default values', () => {
    const wrapper = shallow(<ViewsContainer views={views} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should render 4 views', () => {
    const viewsCount = views.length
    const wrapper = mount(<ViewsContainer views={views} />)
    // Count should be double because slick carousel clones all its slides for infinite scrolling
    expect(wrapper.find('ViewsCard')).toHaveLength(viewsCount * 2)
  })
})
