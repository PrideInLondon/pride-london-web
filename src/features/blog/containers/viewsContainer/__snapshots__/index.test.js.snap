// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewsContainer should render with default values 1`] = `
<section>
  <Gridstyles__Container
    mx="auto"
  >
    <Gridstyles__Row
      display="flex"
      flexWrap="wrap"
    >
      <Gridstyles__Column
        width={1}
      >
        <styles__CardContainer>
          <styles__ViewsTitle
            isCentered={true}
            isLight={true}
          >
            Views
          </styles__ViewsTitle>
          <Slider
            arrows={true}
            infinite={true}
            responsive={
              Array [
                Object {
                  "breakpoint": 767,
                  "settings": Object {
                    "arrows": false,
                    "centerMode": true,
                    "centerPadding": "100px",
                    "slidesToScroll": 1,
                    "slidesToShow": 1,
                  },
                },
                Object {
                  "breakpoint": 550,
                  "settings": Object {
                    "arrows": false,
                    "centerMode": true,
                    "centerPadding": "40px",
                    "slidesToScroll": 1,
                    "slidesToShow": 1,
                  },
                },
              ]
            }
            slidesToScroll={1}
            slidesToShow={3}
            speed={500}
          >
            <ViewsCard
              author={
                Object {
                  "displayName": "Test",
                }
              }
              category="Views"
              datePublished="2019-05-17"
              featuredImage={
                Object {
                  "fixed": Object {
                    "base64": "abc",
                    "height": 400,
                    "src": "src",
                    "srcSet": "srcset",
                    "width": 400,
                  },
                }
              }
              id="1"
              key="1"
              slug="somewhere1"
              title="test"
            />
            <ViewsCard
              author={
                Object {
                  "displayName": "Test 2",
                }
              }
              category="Views"
              datePublished="2019-05-17"
              featuredImage={
                Object {
                  "fixed": Object {
                    "base64": "abc",
                    "height": 400,
                    "src": "src",
                    "srcSet": "srcset",
                    "width": 400,
                  },
                }
              }
              id="2"
              key="2"
              slug="somewhere2"
              title="test"
            />
          </Slider>
        </styles__CardContainer>
      </Gridstyles__Column>
    </Gridstyles__Row>
  </Gridstyles__Container>
</section>
`;
